node-api/backend/blobs.go

```diff
@@ -34,8 +36,23 @@ func (b *Backend) BlobSidecarsByIndices(slot math.Slot, indices []uint64) ([]*ap
 	if currentSlot < 0 {
 		return nil, errors.New("invalid negative block height")
 	}
+
+	// If the requested slot is 0 (head, finalized, justified), use the current slot.
+	if slot == 0 {
+		slot = math.Slot(currentSlot)
+	}
```

Comment:
this is the main fix of this PR. we resolve "head", "finalized" and "justified" blockIDs to slot 0, which hits the node query context for the latest height.